test/python/quantum_info/test_synthesis.py

```diff
@@ -173,6 +174,28 @@ def assertRoundTrip(self, weyl1: TwoQubitWeylDecomposition):
         self.assertEqual(maxdiff, 0, msg=f"K2r matrix differs by {maxdiff}" + msg_base)
         self.assertEqual(weyl1.requested_fidelity, weyl2.requested_fidelity, msg_base)
 
+    def assertRoundTripPickle(self, weyl1: TwoQubitWeylDecomposition):
+        """Fail if loads(dumps(weyl1)) not equal to weyl1"""
```

Comment:
It may also be worth adding a very similar test that `copy.deepcopy` and `copy.copy` work correctly as well/ `deepcopy` presumably will because it's a full pickle/unpickle by default, but `copy` might do something funny; I think it basically calls `__getstate__`, then restores it with an immediate call to `__setstate__` without recursing the pickle through the state.  I would guess it does the right thing and calls `__getnewargs_ex__` if defined, but I don't really know, and it might be worth a test, since Qiskit uses the `copy` module pretty extensively.